components/RefinementList.js

```diff
@@ -19,7 +19,10 @@ class RefinementList extends React.Component {
   // has a checkbox inside, we ignore the first click event because we will get another one.
   handleClick(value, e) {
     if (e.target.tagName === 'A' && e.target.href) {
+      // do not trigger any url change by the href
       e.preventDefault();
+      // do not bubble (so that hierarchical lists are not triggering refine twice)
+      e.stopPropagation();
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent 
https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events

I was thinking preventDefault on the event, clone it, stopPropagation on the original, dispatch the clone from the root of the refinementlist